### DISCUSSION TITLE
Plotly only showing me 7 days of data

### DISCUSSION THREAD

**Discussion by qsnow1** [dbuezas/lovelace-plotly-graph-card#297] [General]
Does the data plotly use get purged with the recorder setting in configuration.yaml?  I can only seem to get 7 days or so of data to show up - regardless of my range filters or panning through the graph.  Ultimately I'd like to keep a couple years worth of temperature readings.

**Comment by dbuezas** [comment]
Search for "statistics" in the readme. The normal history is kept for a limited amount of time, but the statistics are stored forever

**Comment by qsnow1** [comment]
Ah thanks.  I had come across that, but when trying it, it caused one of my graphs to disappear so I thought I was on the wrong track.  

What would cause adding statistis to an entity to not graph the data?  Image attached

![2023-07-19_09-10-46](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/49792639/0a674649-fa6a-42f5-9738-f97c1be9c080)

**Reply by dbuezas** [reply]
The recorder may be configured to not store statistics for that entity or the db is corrupted.
I suggest you try:
* Each period manually
* each statistic param
* the history card to confirm it is not the plotly card

Maybe that leads to an insight.

**Comment by qsnow1** [comment]
Ah yes.  That was helpful.  Thanks for pointing in right direction.  For whatever reason the temp sensor didn't have a state_class attribute.  I had to add it in the customizations.yaml

sensor.snow_pool_water_temperature:
  state_class: measurement

**Reply by dbuezas** [reply]
That's why? I observed this myself last month and didn't figure it out! Thanks
